pallets/ah-migrator/src/staking/nom_pools.rs

```diff
@@ -89,16 +100,76 @@ impl<T: Config> Pallet<T> {
 				pallet_nomination_pools::Metadata::<T>::insert(meta.0, meta.1);
 			},
 			ReversePoolIdLookup { lookups } => {
-				log::debug!(target: LOG_TARGET, "Integrating NomPoolsReversePoolIdLookup: {:?}", &lookups.0);
-				pallet_nomination_pools::ReversePoolIdLookup::<T>::insert(lookups.0, lookups.1);
+				let translated_account = Self::translate_account_rc_to_ah(lookups.0.clone());
+				log::debug!(target: LOG_TARGET, "Integrating NomPoolsReversePoolIdLookup: {} -> {}",
+					lookups.0.to_polkadot_ss58(), translated_account.to_polkadot_ss58());
+				pallet_nomination_pools::ReversePoolIdLookup::<T>::insert(
+					translated_account,
+					lookups.1,
+				);
 			},
 			ClaimPermissions { perms } => {
-				log::debug!(target: LOG_TARGET, "Integrating NomPoolsClaimPermissions: {:?}", &perms.0);
-				pallet_nomination_pools::ClaimPermissions::<T>::insert(perms.0, perms.1);
+				let translated_account = Self::translate_account_rc_to_ah(perms.0.clone());
+				log::debug!(target: LOG_TARGET, "Integrating NomPoolsClaimPermissions: {} -> {}",
+					perms.0.to_polkadot_ss58(), translated_account.to_polkadot_ss58());
+				pallet_nomination_pools::ClaimPermissions::<T>::insert(translated_account, perms.1);
 			},
 		}
 	}
 
+	/// Translate accounts in bonded pool roles from RC to AH format
+	fn translate_bonded_pool_accounts(mut pool: BondedPoolInner<T>) -> BondedPoolInner<T> {
+		// Translate accounts in pool roles using functional approach
+		pool.roles.depositor = Self::translate_account_rc_to_ah(pool.roles.depositor);
+		pool.roles.root = pool.roles.root.map(|acc| Self::translate_account_rc_to_ah(acc));
```

Comment:
```suggestion
		pool.roles.root = pool.roles.root.map(Self::translate_account_rc_to_ah);
```
Even more functional 😋